src/app/CommandHandler.cpp

```diff
@@ -69,6 +69,10 @@ CHIP_ERROR CommandHandler::AllocateBuffer()
 
         mInvokeResponseBuilder.CreateInvokeResponses();
         ReturnErrorOnFailure(mInvokeResponseBuilder.GetError());
+
+        auto * tlvWriter = mInvokeResponseBuilder.GetWriter();
+        ReturnErrorOnFailure(tlvWriter->ReserveBuffer(mInvokeResponseBuilder.GetInvokeResponses().GetSizeToEndInvokeResponses()));
```

Comment:
So this is assuming that the writer `mInvokeResponseBuilder` uses and the writer `mInvokeResponseBuilder.GetInvokeResponses()` are the same writer, right?

That happens to be true now, but nothing guarantees it.  It might be cleaner to reserve on the writer that is being used.  So if we need space for `mInvokeResponseBuilder.GetInvokeResponses().EndOfInvokeResponses()`, that should be reserved and un-reserved on the writer that `mInvokeResponseBuilder.GetInvokeResponses()` has.

To make that easier, I guess we could put reserve/unreserve methods on `InvokeResponseIBs::Builder` and `InvokeResponseMessage::Builder`....

Followup is fine for this part, if desired.